platform/osgi/platform-osgi-conditions/src/test/java/org/codice/ddf/condition/SignerConditionTest.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.condition;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.service.condpermadmin.ConditionInfo;
+import sun.security.x509.X509CertImpl;
+
+public class SignerConditionTest {
```

Comment:
❗️ We're missing many conditions and paths here that would be really difficult to test higher up. We should mock out the value returned by `bundle.getSignerCertificates(Bundle.SIGNERS_TRUSTED)` and test all the paths and cases in `isSatisfied()`, `getExtendedCertAttribute()`, `getSubjectAlternativeNames()` and `collectAltNames()`, otherwise we won't know for sure that this code always works as expected and may cause a hard to find security regression down the road.

Reply:
I added a test for subject alt name, but I'm not sure how to populate it with binary data. I know it is possible and the code there should work.

Reply:
Can't we just use a mock instead of reading in a real `.der` file? Wouldn't that make it possible to test those cases? I'm just worried that if we don't have good unit tests here, someone could introduce a security related bug in the future and we won't know about it.

Reply:
I'd just be testing the mock instead of reading a real der sequence that is actually saved in a cert at that point. Someone could definitely introduce something here. However, we don't really have any reason to use this code in DDF and it is mainly meant for downstream projects if they ever have a need for adding conditions against bundle signers. I could also remove the code that deals with bytes, I'd just rather not do that.

Reply:
Ok, let's not worry about that part then.

Can we add a test for this exception though: https://github.com/codice/ddf/pull/3073/files#diff-48a09791a0210f51250e2c0cdfce43ebR108?

We should also make https://github.com/codice/ddf/pull/3073/files#diff-48a09791a0210f51250e2c0cdfce43ebR125 a `RuntimeException` instead of an `Exception` and see if we can add a unit test for it as well.

Reply:
Yep, added tests for the exceptions.